DOCS-11172 Update Oracle EBS 12.2 Reference (#1498)
mulesoft/docs-connectors: oracle-ebs-122/2.1/modules/ROOT/pages/new-reference.adoc

= Oracle EBS 12.2 Connector Reference - Mule 4
// :page-aliases: connectors::oracle/oracle-ebs-122-connector-reference.adoc

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Premium]

> Oracle EBS 12.2 Connector v2.1

Reviewer: Delete line 6.

Author: I remember we added this on purpose in a previous release to clarify the version since 12.2 is in the name. But if it looks weird or doesn't match the template I can remove it-- what do you think? @valkyrie69

* <<repeatable-file-store-stream>>
* non-repeatable-stream |  Configures how Mule processes streams. The default is to use repeatable streams. |  |
| PL/SQL a| String |  PL/SQL name. |  | x
| Operation a| String |  Operation to execute. |  | x
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
| Target Value a| String |  Expression that evaluates the operation's output. The outcome of the expression is stored in the *Target Variable* field. |  #[payload] |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |Binary
|===

=== For Configurations
* <<config, Default Configuration>>

==== Throws
* ORACLE-EBS122:UNKNOWN
* ORACLE-EBS122:BAD_REQUEST
* ORACLE-EBS122:METADATA_ERROR
* ORACLE-EBS122:RETRY_EXHAUSTED
* ORACLE-EBS122:JMS
* ORACLE-EBS122:CONNECTIVITY
* ORACLE-EBS122:UNAUTHORIZED
* ORACLE-EBS122:INTERNAL_SERVER_ERROR
* ORACLE-EBS122:REQUEST_FAILED
* ORACLE-EBS122:FORBIDDEN
* ORACLE-EBS122:XML_PARSING

== Sources

[[BusinessEventsSource]]
=== Business Events Source
`<oracle-ebs122:business-events-source>`

[NOTE]

> The *Business Events Source* does not filter the incoming events. The user must filter the events after they arrive to the source. +

Reviewer: ```suggestion
The *Business Events Source* does not filter the incoming events. The user must filter the events after they arrive at the source. +
```

Reviewer: You might want to double-check the source name. Not sure if it's actually called *Business Events Source*, or if it's *Business Events* source.

Reviewer 2: It shows in studio as "Business Events Source".

| Client Timeout a| Number |  |  |
| Send Tcp No Delay a| Boolean |  | true |
| Linger a| Number |  |  |
| Keep Alive a| Boolean |  | false |
| Connection Timeout a| Number |  | 30000 |
|===

[[Tls]]
=== TLS

Configures TLS to provide secure communications for the Mule app.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Enabled Protocols a| String | Comma-separated list of protocols enabled for this context. |  |
| Enabled Cipher Suites a| String | Comma-separated list of cipher suites enabled for this context. |  |
| Trust Store a| <<TrustStore>> | Configures the TLS truststore. |  |
| Key Store a| <<KeyStore>> | Configures the TLS keystore. |  |
| Revocation Check a| * <<standard-revocation-check>>
* <<custom-ocsp-responder>>
* <<crl-file>> | Configures how to validate certificates. |  |
|===

[[TrustStore]]
=== Truststore

Configures the truststore for TLS.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Path a| String | Path to the truststore. Mule resolves the path relative to the current classpath and file system, if possible. |  |

> | Password a| String | Password used to protect the trust store. |  |

Reviewer: ```suggestion
| Password a| String | Password used to protect the truststore. |  |
```

| Type a| String | Type of truststore. |  |
| Algorithm a| String | Encryption algorithm that the truststore uses. |  |
| Insecure a| Boolean | If `true`, Mule stops performing certificate validations. Settig this to `true` can make connections vulnerable to attacks. |  |
|===

[[KeyStore]]
=== Keystore

Configures the keystore for the TLS protocol. The keystore you generate contains a private key and a public certificate.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Path a| String | Path to the keystore. Mule resolves the path relative to the current classpath and file system, if possible. |  |

> | Type a| String | Type of store used. |  |

Reviewer: ```suggestion
| Type a| String | Type of keystore used. |  |
```
